functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,192 @@
+        rethrpc = reth.create_rpc()
+
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+        el_address = web3.address
+        self.logger.debug(f"EL address: {el_address}")
+
+        # Get the balance of the EL address prior to the deposits
+        # FIXME: I have no idea why the initial balance is so high
+        balance_pre = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance before deposits: {balance_pre}")
+        assert (
+            balance_pre == ROLLUP_DEPOSIT_AMOUNT * SATS_TO_WEI * 100_000
+        ), "Strata balance is not expected"
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        bridge_pk = get_bridge_pubkey(seqrpc)
+
+        # Generate Plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Deposit to the EL address
+        # NOTE: we need 2 deposits to make sure we have funds for gas
+        self.make_drt(ctx, el_address, bridge_pk)
+        self.make_drt(ctx, el_address, bridge_pk)
+        time.sleep(0.5)
+
+        # Get the balance of the EL address prior to the withdrawal
+        balance_post = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance after deposits: {balance_post}")
+        assert balance_post > 0, "Strata balance is not greater than 0"
+
+        # Send funds to the bridge precompile address for a withdrawal
+        change_address_pk = extract_p2tr_pubkey(withdraw_address)
+        self.logger.debug(f"Change Address PK: {change_address_pk}")
+        l2_tx_hash = self.make_withdraw(ctx, change_address_pk, GAS).hex()
+        self.logger.debug(f"Sent withdrawal transaction with hash: {l2_tx_hash}")
+
+        # Wait for the withdrawal to be processed
+        time.sleep(DEFAULT_BLOCK_TIME_SEC * 2)
+
+        l2_tx_state = web3.eth.get_transaction_receipt(l2_tx_hash)
+        self.logger.debug(f"Transaction state (web3): {l2_tx_state}")
+
+        # Make sure that the balance is less than the initial balance
+        balance_post_withdraw = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance after withdrawal: {balance_post_withdraw}")
+        difference = balance_post - balance_post_withdraw
+        self.logger.debug(f"Strata Balance difference: {difference}")
+        # NOTE: Difference is D BTC - gas_fee
+        assert difference == 10_000_023_000_818_888_144, "balance difference is not expected"
```

Comment:
It's more ergonomic to have descriptive constants here rather than comments. Right now, the tests are failing at this point and the error message just prints this line out (without the comment). We'd know exactly what failed if this line said: `assert difference == DEPOSIT_AMOUNT - WITHDRAWAL_GAS_FEE` instead.

You might find this interesting: https://youtu.be/Bf7vDBBOBUA?si=kQQbb18nJ69MCplM

Reply:
Agreed. Much cleaner and future proof to define constants and use arithmetic expressions where needed (unless the expression evaluation itself were to be prohibitively expensive, which it isn't here).

Reply:
Rajil thanks for the review, and yes you're totally correct.
I will mark this as a draft and work on it this week.
Will mark as ready once I think is in a bulletproof state to not waste review time.
Thanks!